mavros/src/plugins/setpoint_trajectory.cpp

```diff
@@ -123,10 +137,21 @@ class SetpointTrajectoryPlugin : public plugin::PluginBase,
 		float yaw, yaw_rate;
 		uint16_t type_mask;		
 		if(!setpoint_target->transforms.empty()){
-			position = ftf::to_eigen(setpoint_target->transforms[0].translation);
-			tf::quaternionMsgToEigen(setpoint_target->transforms[0].rotation, attitude);
-			yaw = ftf::quaternion_get_yaw(ftf::transform_orientation_enu_ned(
-					ftf::transform_orientation_baselink_aircraft(attitude)));
+			position = [&] () {
+				if (static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_NED || static_cast<MAV_FRAME>(mav_frame) == MAV_FRAME::BODY_OFFSET_NED) {
+					return ftf::transform_frame_baselink_aircraft(ftf::to_eigen(setpoint_target->transforms[0].translation));
+				} else {
+					return ftf::transform_frame_enu_ned(ftf::to_eigen(setpoint_target->transforms[0].translation));
+				}
+			} ();
```

Comment:
https://github.com/mavlink/mavros/blob/master/mavros/include/mavros/frame_tf.h#L180

I think better just select StaticTF once.

Reply:
Once you mean when we receive the trajectory message?

Reply:
I'm a bit aware where it's better to place, but in my opinion, you need only once select transform type (baselink-aircraft / enu-ned) and then just use `ftf::transform_frame()` / `ftf::transform_orientation()`.
That also will look much better than current lambdas.